backend/src/api/model/block/mutations.rs

```diff
@@ -63,6 +63,35 @@ impl BlockValue {
             .ok_or_else(|| invalid_input!("`realm` does not refer to a valid realm"))
     }
 
+    pub(crate) async fn add_video(
+        realm: Id,
+        index: i32,
+        block: NewVideoBlock,
+        context: &Context,
+    ) -> ApiResult<Realm> {
+        context.require_moderator()?;
+
+        let (realm, index) = Self::prepare_realm_for_block(realm, index, context).await?;
+
+        context.db
+            .execute(
+                "insert into blocks (realm_id, index, type, title, video_id) \
+                    values ($1, $2, 'video', $3, $4)",
+                &[
+                    &realm,
+                    &index,
+                    &block.title,
+                    &block.event.key_for(Id::EVENT_KIND)
+                        .ok_or_else(|| invalid_input!("`block.event` does not refer to an event"))?,
```

Comment:
If you store this in a variable, you could put all the arguments for the query in a single line... :shrug: